pudb/debugger.py

```diff
@@ -403,6 +403,7 @@ def interaction(self, frame, exc_tuple=None, show_exc_dialog=True):
         self.stack, index = self.get_shortened_stack(frame, tb)
 
         if self.post_mortem:
+            self.stack.append((self.bottom_frame, self.bottom_frame.f_lineno))
```

Comment:
How does `pdb` handle this situation? `self.stack` here is something that's managed by `bdb`, I'm not sure we're at liberty to mess with it.

Reply:
After finishing the execution of the script, pdb enters [this finally block](https://github.com/python/cpython/blob/e5862f79c16e28f1ec51d179698739a9b2d8c1d2/Lib/bdb.py#L600) (kinda like our post-mortem mode). If at this point, we type `p <variable name>` on the command line, pdb will print a `NameError`, and the user continues with the pdb REPL. So currently our `IndexError` is similar to this `NameError`.

I haven't dig into how bdb manages the stack in this case. 

The problem with the current code is that if I have an object at the module level, like `a = list(range(100))` for example, I can't expand it to see what the elements are in post-mortem. This PR allows user to inspect data structures like that.

Reply:
Instead of modifying data structures that aren't ours to modify, could you instead create a `@property` that provides a modified view of the stack, and then change the relevant spots of the code to use that view of the stack instead? Also make sure to add comments to explain why this is being done.